src/rsqsim_api/rsqsim_api/catalogue/catalogue.py

```diff
@@ -463,12 +489,244 @@ def events_by_number(self, event_number: Union[int, np.int, Iterable[np.int]], f
+        self.accumulated_slip = {}
         for patch_i in np.unique(self.patch_list):
             matching = (self.patch_list == patch_i)
             accumulated_slip = self.patch_slip[matching].sum()
-            fault_model.patch_dic[patch_i].total_slip = accumulated_slip
+            self.accumulated_slip[patch_i] = accumulated_slip
+
+    def assign_event_mean_slip(self,fault_model: RsqSimMultiFault):
+        """
+        Create dict of event ids with associated mean slip on the patches which slip in them.
+        Note that this overwrites any other value which could have been assigned to mean slip.
+        """
+        self.event_mean_slip = {}
+        for event in self.all_events(fault_model):
+            event.find_mean_slip()
+            self.event_mean_slip[event.event_id] = event.mean_slip
+
+
+
+    def plot_accumulated_slip_2d(self, fault_model: RsqSimMultiFault, subduction_cmap: str = "plasma",
+                                 crustal_cmap: str = "viridis", show: bool = True,
+                                 write: str = None, subplots=None, global_max_sub_slip: int = 0,
+                                 global_max_slip: int = 0,
+                                 figsize: tuple = (6.4, 4.8), hillshading_intensity: float = 0.0, bounds: tuple = None,
+                                 plot_rivers: bool = True, plot_lakes: bool = True,
+                                 plot_highways: bool = True, plot_boundaries: bool = False,
+                                 create_background: bool = False,
+                                 coast_only: bool = True, hillshade_cmap: colors.LinearSegmentedColormap = cm.terrain,
+                                 plot_log_scale: bool = False, log_cmap: str = "magma", log_min: float = 1.0,
+                                 log_max: float = 100., plot_traces: bool = True, trace_colour: str = "pink",
+                                 min_slip_percentile: float = None, min_slip_value: float = None,
+                                 plot_zeros: bool = True):
+        # TODO: Plot coast (and major rivers?)
+        assert self.accumulated_slip is not None, "Need accumulated slip to plot!"
+        if bounds is None and fault_model.bounds is not None:
+            bounds = fault_model.bounds
+
+        if all([min_slip_percentile is not None, min_slip_value is None]):
+            min_slip = np.percentile(self.patch_slip, min_slip_percentile)
+        else:
+            min_slip = min_slip_value
+
+        if subplots is not None:
+            if isinstance(subplots, str):
+                # Assume pickled figure
+                with open(subplots, "rb") as pfile:
+                    loaded_subplots = pickle.load(pfile)
+                fig, ax = loaded_subplots
+            else:
+                # Assume matplotlib objects
+                fig, ax = subplots
+        elif create_background:
+            #TODO: add this directory + file to repo data/other_lines/nz-lake-polygons-topo-1250k.shp
```

Comment:
Did you add this or was it there already? I think I've done this.

Reply:
I added it because I got an error saying it wasn't there (and I can't find it)